plugins/feedback.py

```diff
@@ -0,0 +1,169 @@
+        """
+        authorid = None
+        if self.author is not None:
+            authorid = self.author.id
+
+        return {
+            "id": self.id,
+            "authorid": authorid,
+            "msglink": self.msg_link,
+            "content": self.content,
+        }
+
+    @classmethod
+    def deserialize(cls, bot, cid, d):
+        """
+        Constructs a Complaint object from a dict.
+        :param bot: Geckarbot reference
+        :param cid: Complaint id
+        :param d: dict made by serialize()
+        :return: Complaint object
+        """
+        author = discord.utils.get(bot.guild.members, id=d["authorid"])
+        return Complaint(cid, author, d["msglink"], d["content"])
+
+    @classmethod
+    def from_message(cls, plugin, msg):
+        content = msg.content[len("!complain"):].strip()  # todo check if necessary
+        return cls(plugin.get_new_id(), msg.author, msg.jump_url, content)
+
+    def to_message(self):
+        authorname = "Not found"
+        if self.author is not None:
+            authorname = utils.get_best_username(self.author)
+        r = "**#{}**: {}: {}".format(self.id, authorname, self.content)
+        if self.msg_link is not None:
+            r += "\n{}".format(self.msg_link)
+        return r
+
+
+class Plugin(BasePlugin):
+    def __init__(self, bot):
+        super().__init__(bot)
+        bot.register(self)
+
+        self.storage = Config().get(self)
+        self.complaints = {}
+        self.highest_id = None
+
+        # Load complaints from storage
+        if self.storage is None:
+            self.storage = deepcopy(skeleton_config)
+        else:
+            str_keys_to_int(self.storage["complaints"])
+        for cid in self.storage["complaints"]:
+            self.complaints[cid] = Complaint.deserialize(self.bot, cid, self.storage["complaints"][cid])
+
+        self.get_new_id(init=True)
+
+    def get_new_id(self, init=False):
+        """
+        Acquires a new complaint id
+        :param init: if True, only sets self.highest_id but does not return anything. Useful for plugin init.
+        :return: free unique id that can be used for a new complaint
+        """
+        if self.highest_id is None:
+            self.highest_id = 0
+            for el in self.complaints:
+                if el > self.highest_id:
+                    self.highest_id = el
+        if not init:
+            self.highest_id += 1
+            return self.highest_id
+
+    def write(self):
+        r = deepcopy(skeleton_config)
+        for el in self.complaints:
+            complaint = self.complaints[el]
+            r["complaints"][complaint.id] = complaint.serialize()
+        Config().set(self, r)
+        Config().save(self)
+
+    @commands.command(name="redact", help="Redacts the list of complaits (i.e. read and delete)")
+    async def redact(self, ctx, *args):
+        if not permChecks.check_full_access(ctx.author):
+            raise commands.MissingAnyRole
```

Comment:
`@commands.has_any_role(Config().ADMIN_ROLE_ID, Config().BOTMASTER_ROLE_ID)`
macht exakt das gleiche, nur in besser